processor/file_test.go

```diff
@@ -85,8 +84,10 @@ func TestWalkDirectoryParallel(t *testing.T) {
 	GcFileCount = 10
 
 	inputChan := make(chan *FileJob, 10000)
-	walkDirectoryParallel("../", inputChan)
-	close(inputChan)
+
+	dirwalker := NewDirectoryWalker(inputChan)
+	dirwalker.Walk("../")
+	dirwalker.Run()
```

Comment:
Maybe return the walker on the walk? Could then chain it.

```
dirwalker := NewDirectoryWalker(inputChan).Walk("../").Run()
```

Reply:
That wouldn't work because `Walk` actually returns an error that was being ignored. I've updated the tests to check the returned error value.